relay-server/src/actors/events.rs

```diff
@@ -1586,11 +1586,16 @@ impl Handler<HandleEnvelope> for EventManager {
                             .body(
                                 envelope
                                     .to_vec()
+                                    // XXX: upstream actor should allow for custom error type,
+                                    // right now we are forced to shoehorn our envelope errors into
+                                    // UpstreamRequestError
                                     .map_err(failure::Error::from)
-                                    .map_err(actix_web::Error::from)?
+                                    .map_err(actix_web::Error::from)
+                                    .map_err(HttpError::Actix)
+                                    .map_err(UpstreamRequestError::Http)?
```

Comment:
_nit_: Refactor the `envelope.to_vec()` chain into its own variable and then move `builder.body()` together with the previous builder chain.